### PR TITLE
fix: mark unlisted NuGet package version as deprecated

### LINKED ISSUE #7299: Renovate creates PRs for unlisted NuGet package versions
**What Renovate type, platform and version are you using?**

GitHub (happens everywhere, we discovered it on self-hosted GitLab)

**Describe the bug**

Renovate tries to update to unlisted Nuget package version, probably since https://github.com/renovatebot/renovate/pull/6943 was merged (/cc @zharinov @viceice)

**To Reproduce**

- Add a reference to a NuGet package that has a higher version that was unlisted (e.g. https://www.nuget.org/packages/System.ComponentModel.Composition where https://www.nuget.org/packages/System.ComponentModel.Composition/2010.2.11.0 is unlisted).  See https://github.com/fgreinacher/renovate-include-consumer/blob/master/test.csproj
- Wait for Renovate to create a PR for new unlisted version. See
https://github.com/fgreinacher/renovate-include-consumer/pull/6/files

**Additional hints**

- Renvoate should probably filter `listed: false` entries from https://api.nuget.org/v3/registration5-semver1/system.componentmodel.composition/index.json

### PR DESCRIPTION
**Pull request by fgreinacher**
Fixes #7299

### CODE REVIEW

File: lib/datasource/nuget/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -323,6 +323,7 @@ Object {
       "version": "2.5.10.11092",
     },
     Object {
+      "isDeprecated": true,
```
Comment (fgreinacher) [inline lib/datasource/nuget/__snapshots__/index.spec.ts.snap:326]: The existing test fixtures contained several unlisted/deprecated packages, so I deliberately did not write additional tests.

**Comment by fgreinacher** [comment]
Windows test failed with timeout (https://github.com/renovatebot/renovate/pull/7304/checks?check_run_id=1123806323#step:13:199). Could someone retrigger the job?

**Comment by rarkins** [comment]
@fgreinacher unfortunately Actions' cache is being a pain again. Let's see how it goes..

**Comment by fgreinacher** [comment]
Checks are now passing. Ready for review 👍

**Review by rarkins — APPROVED** [review]

**Pull request merged**